src/components/ColorPicker/components/ColorPickerContent/ColorPickerContentComponent.jsx

```diff
@@ -2,15 +2,102 @@ import cx from "classnames";
 import _difference from "lodash/difference";
 import _intersection from "lodash/intersection";
 import PropTypes from "prop-types";
-import React, { useCallback, useMemo } from "react";
+import React, { useCallback, useMemo, useRef } from "react";
 import { SIZES } from "../../../../constants/sizes";
 import { COLOR_STYLES, contentColors } from "../../../../general-stories/colors/colors-vars-map";
 import Button from "../../../Button/Button";
 import NoColor from "../../../Icon/Icons/components/NoColor";
 import ColorPickerItemComponent from "../ColorPickerItemComponent/ColorPickerItemComponent";
-import { DEFAULT_NUMBER_OF_COLORS_IN_LINE } from "../../ColorPickerConstants";
+import { COLOR_SHAPES, DEFAULT_NUMBER_OF_COLORS_IN_LINE } from "../../ColorPickerConstants";
 import { calculateColorPickerWidth } from "../../services/ColorPickerStyleService";
 import "./ColorPickerContentComponent.scss";
+import useGridKeyboardNavigation from "../../../../hooks/useGridKeyboardNavigation/useGridKeyboardNavigation";
+import {
+  GridKeyboardNavigationContext,
+  useGridKeyboardNavigationContext
+} from "../../../GridKeyboardNavigationContext/GridKeyboardNavigationContext";
+
+const Colors = React.forwardRef(
```

Comment:
Both this and `ClearButton` take up quite a bit of this file, consider separating them to dedicated files.

Reply:
good idea. I split it.
as a more general question - do we usually add PropTypes to such "internal" components which are used only in a specific component? (I haven't used PropTypes so much in the past)

Reply:
Yes, I think that's the best practice, even though not all components implement it fully.